observability/braintrust/src/tracing.ts

```diff
@@ -109,13 +138,21 @@ export class BraintrustExporter extends BaseExporter {
 
     const payload = this.buildSpanPayload(span);
 
+    // When attaching to an external parent (eval/logger span), don't pass Mastra's internal
+    // parentSpanIds. Let Braintrust auto-handle the parent-child relationship.
+    const shouldOmitParentIds = spanData.isExternal && !span.parentSpanId;
+
     const braintrustSpan = braintrustParent.startSpan({
       spanId: span.id,
       name: span.name,
       type: mapSpanType(span.type),
-      parentSpanIds: span.parentSpanId
-        ? { spanId: span.parentSpanId, rootSpanId: span.traceId }
-        : { spanId: span.traceId, rootSpanId: span.traceId },
+      ...(shouldOmitParentIds
+        ? {} // Let Braintrust auto-link to parent
+        : {
+            parentSpanIds: span.parentSpanId
+              ? { spanId: span.parentSpanId, rootSpanId: span.traceId }
+              : { spanId: span.traceId, rootSpanId: span.traceId },
+          }),
```

Comment:
I think this could be an issue.  We currently return a `traceId` from agent/workflow runs that users can use to search for the trace in Braintrust and other exporters.

If the mastra `traceId` isn't in the root-level span, can the user still find the trace?